src/api/register_request.rs

```diff
@@ -154,23 +173,21 @@ fn check_port_not_available<T>(
     register_input_error: &Result<T, RegisterInputError>,
     port: u16,
 ) -> Result<(), ApiError> {
-    if let Err(RegisterInputError::DecoderError(ref id, InputInitError::InputError(ref err))) =
-        register_input_error
-    {
-        if let Some(err) = err.0.downcast_ref::<std::io::Error>() {
-            match err.kind() {
-                std::io::ErrorKind::AddrInUse =>
-                    Err(ApiError::new(
-                        PORT_ALREADY_IN_USE_ERROR_CODE,
-                        format!("Failed to register input stream \"{id}\". Port {port} is already in use or not available."),
-                        tiny_http::StatusCode(400)
-                    )),
-                _ => Ok(())
-            }
-        } else {
-            Ok(())
-        }
-    } else {
-        Ok(())
+    let Err(RegisterInputError::InputError(ref id, err)) = register_input_error else {
```

Comment:
Now the pipeline crate is aware of RTP shouldn't this be a part of pipeline implementation? no strong opinion, just sth to consider